pkg/executor/core/executor.go

```diff
@@ -156,3 +160,27 @@ func (ex Executor) verifyReference(ctx context.Context, subjectRef common.Refere
 
 	return types.VerifyResult{IsSuccess: isSuccess, VerifierReports: verifyResults}
 }
+
+func (ex Executor) addNestedVerifierResult(ctx context.Context, referenceDesc ocispecs.ReferenceDescriptor, subjectRef common.Reference, verifyResult *vr.VerifierResult) {
+	verifyParameters := e.VerifyParameters{
+		Subject:        fmt.Sprintf("%s@%s", subjectRef.Path, referenceDesc.Digest),
+		ReferenceTypes: []string{"*"},
+	}
+
+	// error is never returned from ex.VerifySubject, if an err occurs, the result contains the error message
+	nestedVerifyResult, err := ex.VerifySubject(ctx, verifyParameters)
```

Comment:
Hi , A question for the updated implementation , how do we enforce something like "all SBOM must have signature"?

Reply:
For the updated implementation, you would need to set 
```json
{
    "executor": {
        "nestedReferences":{
            "org.example.sbom.v0": true
        }
    },
}
```

This would enable to executor to fetch all nested reference for artifact types of `org.example.sbom.v0`. To enforce all SBOMs must have signatures, you would need to do that though at the gatekeeper policy level.

The current implementation sets `nestedReferences` as a comma separated string of artifact types rather than a boolean. However, there is no effective behavior change since the ORAS store doesn't do anything with this anyway, meaning it does not filter for artifact types, it just gets everything at the next level.

Reply:
thanks for the details. Just curious, does the verifier configuration enforce "all SBOM must have signature" today? If nested signature type  is configured today for SBOM verifier, when no signature were found on SBOM, is result a failure?

Reply:
I should have said that it is enforced at the policy level (not just gatekeeper). Setting the following should enforce signatures for both this implementation and the current one as long as the nestedReferences field is also set. You need to have both
```json
      "artifactVerificationPolicies": {
        "application/vnd.cncf.notary.signature": "any" // or all depending on the use case
      }
```

Reply:
ah got it. thanks for the clarification!